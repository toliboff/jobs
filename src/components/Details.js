import React from 'react';
import { useSelector } from 'react-redux';
import { BiRightArrowCircle } from 'react-icons/bi';
import Header from './Header';

const Details = () => {
  const state = useSelector((state) => state.companiesReducer);
  return (
    <div>
      <Header title="company" count={state.totalCap} />
      <ul className="companies">
        {state.companies.map((company) => (
          <li key={company.symbol} className="company">
            <span>{company.name}</span>
            <div className="companyCap">
              $
              {company.marketCap}
              <span>&nbsp;billion</span>
              <BiRightArrowCircle className="arrow" />
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Details;
