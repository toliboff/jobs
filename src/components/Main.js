import React from 'react';
import { NavLink } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { BiRightArrowCircle } from 'react-icons/bi';
import Header from './Header';
import { fetchCompanies } from '../redux/companies/companies';
import images from '../assets/images';

const Main = () => {
  const dispatch = useDispatch();

  const selectMarket = (market) => {
    dispatch(fetchCompanies(market));
  };

  const state = useSelector((state) => state.stockmarketReducer);
  return (
    <div data-testid="main-component">
      <Header title="market" count={state.totalCap} filter="STOCK" />
      <ul className="categories">
        {state.markets.map((market, index) => (
          <li
            key={market.name}
            className={index % 2 === 0 ? 'category' : 'category dark-bg'}
            style={{ backgroundImage: `url('${images[market.name]}')` }}
          >
            <NavLink to="/details" onClick={() => selectMarket(market.name)} className="link">
              <BiRightArrowCircle />
              <div className="exchange">
                <h2>{market.name}</h2>
                <div className="marketCap">
                  $
                  {market.marketCap}
                  B
                </div>
              </div>
            </NavLink>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Main;
