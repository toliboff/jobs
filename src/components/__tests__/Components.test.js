import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/configStore';
import Main from '../Main';
import Details from '../Details';

afterEach(() => {
  cleanup();
});

test('should render Main component', () => {
  render(<Provider store={store}><Router><Main /></Router></Provider>);
  const mainElement = screen.getByTestId('main-component');
  expect(mainElement).toBeInTheDocument();
  expect(mainElement).toHaveTextContent('market capitalisation');
});

test('should render Details component', () => {
  render(<Provider store={store}><Router><Details /></Router></Provider>);
  const detailsElement = screen.getByTestId('details-component');
  expect(detailsElement).toHaveTextContent('company capitalisation');
});

test('matches snapshot', () => {
  const tree = renderer
    .create(<Provider store={store}><Router><Details /></Router></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
