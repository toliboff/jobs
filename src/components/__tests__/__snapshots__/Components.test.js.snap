// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="details-component"
>
  <div
    className="header"
  >
    <div
      className="navigation"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          className="controls"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </a>
      <span>
        company
         capitalisation
      </span>
      <div>
        <svg
          className="controls"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 320c37.712 0 68.571-30.924 68.571-68.714V100.714C324.571 62.924 293.712 32 256 32s-68.571 30.924-68.571 68.714v150.572c0 37.79 30.859 68.714 68.571 68.714zm121.139-75.452c0 68.714-58.282 116.815-121.139 116.815s-121.139-48.102-121.139-116.815H96c0 77.873 61.719 143.153 137.144 153.465V480h45.713v-81.987C354.281 386.561 416 322.421 416 244.548h-38.861z"
          />
        </svg>
        <svg
          className="controls"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 14 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 8.77v-1.6l-1.94-.64-.45-1.09.88-1.84-1.13-1.13-1.81.91-1.09-.45-.69-1.92h-1.6l-.63 1.94-1.11.45-1.84-.88-1.13 1.13.91 1.81-.45 1.09L0 7.23v1.59l1.94.64.45 1.09-.88 1.84 1.13 1.13 1.81-.91 1.09.45.69 1.92h1.59l.63-1.94 1.11-.45 1.84.88 1.13-1.13-.92-1.81.47-1.09L14 8.75v.02zM7 11c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="metrics-title"
      style={
        Object {
          "backgroundImage": "url('undefined')",
        }
      }
    >
      <div />
      <div>
        <h1>
          Stock
        </h1>
        <div
          className="metrics"
        >
          $
          <span
            className="count"
          >
            
          </span>
          <span>
            B
          </span>
        </div>
      </div>
    </div>
    <div
      className="stats"
    >
      STATS BY 
      company
    </div>
  </div>
  <ul
    className="companies"
  />
</div>
`;
